include/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics.hpp

```diff
@@ -103,13 +148,25 @@ class Dynamics
     /// @param              [in] t A step duration from anInstant to the next in seconds
     /// @param              [in] aDynamicsArray A array of shared pointers to dynamics
     /// @param              [in] anInstant An instant
+    /// @param              [in] aFrame The reference frame in which dynamic equations are resolved
+    /// @param              [in] readIndexes An array containing read coordinates subsets indexes and sizes
+    /// @param              [in] writeIndexes An array containing write coordinates subsets indexes and sizes
 
     static void DynamicalEquations(
         const NumericalSolver::StateVector& x,
         NumericalSolver::StateVector& dxdt,
         const double& t,
-        const Array<Shared<Dynamics>>& aDynamicsArray,
-        const Instant& anInstant
+        const Array<Context>& aContextArray,
+        const Instant& anInstant,
+        const Shared<const Frame>& aFrame
```

Comment:
minor: 
@vishwa2710 I think we were chatting about this the other day, but there should be no need to pass a pointer by reference, right? Since the memory footprint of the reference and the pointer should be identical.
```suggestion
        const Shared<const Frame> aFrame
```